MekHQ/src/mekhq/gui/CampaignGUI.java

```diff
@@ -2490,11 +2490,7 @@ public void handleMissionChanged(MissionEvent ev) {
 
     @Subscribe
     public void handlePersonUpdate(PersonEvent ev) {
-        // only bother recalculating AtB parts availability if a logistics admin has been changed
-        // refreshPartsAvailability cuts out early with a "use AtB" check so it's not necessary here
-        if (ev.getPerson().hasRole(PersonnelRole.ADMINISTRATOR_LOGISTICS)) {
-            refreshPartsAvailability();
-        }
+        refreshPartsAvailability();
```

Comment:
Is the efficiency shortcut that was removed present in refreshPartsAvailability()? Otherwise, that's going to be a lot of pointless refreshing.

Reply:
The check itself is the bug as it is not based on the campaign options and thus ignores the proper availability checks.

Reply:
The question still stands, does the parts availability refresh cut out early if the person doesn't have the correct role (whatever that is)?

Reply:
It's filtered properly thanks to your code (from a quick blame check)

Reply:
I misunderstood what you meant last night, my apologies for that.

I see what you mean now, and... yeah, my fix to handle the campaign option would definitely cause performance issues.

I'm going to convert this to draft as I'll need to investigate the best way to tell the correct roles to actually parse for.